### PR TITLE
Update test_automated_recovery_from_failed_nodes_proactive_IPI

### PR DESCRIPTION
**Pull request by PrasadDesala**
Updated test_automated_recovery_from_failed_nodes_proactive_IPI.py to have,
*.  DC based app pods for running IO.
*.  Select common app pod and OSD running node for inducing failure.
*. DC pod validation post failure
*. Added polarion test cases

Signed-off-by: Prasad Desala <tdesala@redhat.com>

### CODE REVIEW

File: tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_proactive_IPI.py [hunk 1]
```diff
@@ -67,25 +123,36 @@ def test_automated_recovery_from_failed_nodes_IPI_proactive(
...
                         resource=pod_obj, state=constants.STATUS_RUNNING,
                         timeout=200
                     )
-                # 'rook-ceph-crashcollector' on the failed node stucks at pending
-                # state. BZ 1810014 tracks it.
-                # Ignoring 'rook-ceph-crashcollector' pod health check as WA and
-                # deleting its deployment so that the pod disappears
-                # Will revert this WA once the BZ is fixed
                 except ResourceWrongStatusException:
+                    # 'rook-ceph-crashcollector' on the failed node stucks at
```
Comment (ebenahar) [inline tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_proactive_IPI.py]: @PrasadDesala do we have an issue for that?

Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_proactive_IPI.py]: Yes.  
BZ - https://bugzilla.redhat.com/show_bug.cgi?id=1810014
ocs-ci issue - #1631

File: tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_proactive_IPI.py [hunk 2]
```diff
@@ -36,28 +51,69 @@ def init_sanity(self):
...
+        label_worker_node(
+            osd_running_nodes, label_key='dc', label_value='fedora'
+        )
+
+        # Create DC app pods
+        log.info("Creating DC based app pods")
+        if interface == 'rbd':
+            interface = constants.CEPHBLOCKPOOL
+        elif interface == 'cephfs':
+            interface = constants.CEPHFILESYSTEM
```
Comment (RazTamir) [inline tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_proactive_IPI.py]:
```
interface = constants.CEPHBLOCKPOOL if interface == 'rbd' else constants.CEPHFILESYSTEM
```

Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_proactive_IPI.py]: Done

File: tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_proactive_IPI.py [hunk 3]
```diff
@@ -36,28 +51,67 @@ def init_sanity(self):
...
+        osd_running_nodes = get_osd_running_nodes()
+        log.info(f"OSDs are running on nodes {osd_running_nodes}")
+        # Label osd nodes with fedora app
+        label_worker_node(
+            osd_running_nodes, label_key='dc', label_value='fedora'
+        )
+
+        # Create DC app pods
+        log.info("Creating DC based app pods")
+        interface = constants.CEPHBLOCKPOOL if interface == 'rbd' \
```
Comment (ebenahar) [inline tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_proactive_IPI.py:84]: please break lines with parentheses

Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_proactive_IPI.py]: Done

**Comment by ebenahar** [comment]
@PrasadDesala verified?

**Comment by PrasadDesala** [comment]
> @PrasadDesala verified?

yes, it is verified.
PR validation Job PASSED.
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-trigger-test-pr/1420//testReport

**Pull request merged**